src/python/pants/option/global_options.py

```diff
@@ -81,14 +81,25 @@ class ExecutionOptions:
     remote_execution_process_cache_namespace: Any
     remote_instance_name: Any
     remote_ca_certs_path: Any
-    remote_oauth_bearer_token_path: Any
     remote_execution_extra_platform_properties: Any
-    remote_execution_headers: Any
+    remote_execution_headers: Union[None, List[Tuple[str, str]], Callable[[Optional[str]], dict]]
```

Comment:
And I got the type wrong here for the `Callable`. Will fix in next iteration. mypy will probably fail.

Reply:
It would be better to return a data class so we can override more than headers.
so a plugin would return a data class.
initially this dataclass will only have a headers field, but down the line we could add support for overriding other values (such as host name, path, and timeout)

Reply:
Makes sense. Would you want the plugin to have read-only access to the host/path/timeout stuff for now?

Reply:
in the future it will be the plugin that sets those values for the pants remoting client.
For now, I am just adocating to have a dataclass as a place holder (with a single attribute for headers).
the remoting client will only look at this field.
but having this interface in place will make it easier to add stuff in the future (like host, path, timeout, etc...)
without changing stuff in many places.